src/main/java/org/opensearch/knn/index/vectorvalues/KNNVectorValuesFactory.java

```diff
@@ -72,8 +108,9 @@ public static <T> KNNVectorValues<T> getVectorValues(final FieldInfo fieldInfo,
         return getVectorValues(FieldInfoExtractor.extractVectorDataType(fieldInfo), vectorValuesIterator);
     }
 
+    @VisibleForTesting
     @SuppressWarnings("unchecked")
-    private static <T> KNNVectorValues<T> getVectorValues(
+    static <T> KNNVectorValues<T> getVectorValues(
```

Comment:
why to make it package private?

Reply:
Needed for [unit test](https://github.com/opensearch-project/k-NN/pull/2135/files#diff-d70259d4ce2116e0b254c69590c9fd2bec228e43e4bd0eec20d7d38693250b0fR128)

Reply:
lets try to avoid this and use one of the public function to test this function.